qa/L0_shared_memory/test.sh

```diff
@@ -62,32 +63,34 @@ for i in \
         fi
 
         export CLIENT_TYPE=$client_type
-        echo "Test: $i, client type: $client_type" >>$CLIENT_LOG
+        TMP_CLIENT_LOG="./tmp_client.log"
+        echo "Test: $i, client type: $client_type" >>$TMP_CLIENT_LOG
 
         set +e
-        python $SHM_TEST SharedMemoryTest.$i >>$CLIENT_LOG 2>&1
+        python $SHM_TEST SharedMemoryTest.$i >>$TMP_CLIENT_LOG 2>&1
         if [ $? -ne 0 ]; then
+            cat $TMP_CLIENT_LOG
             echo -e "\n***\n*** Test Failed\n***"
             RET=1
         else
             check_test_results $TEST_RESULT_FILE 1
             if [ $? -ne 0 ]; then
-                cat $CLIENT_LOG
+                cat $TEST_RESULT_FILE
                 echo -e "\n***\n*** Test Result Verification Failed\n***"
                 RET=1
             fi
         fi
-        set -e
-
+        cat $TMP_CLIENT_LOG >>$CLIENT_LOG
+        rm $TMP_CLIENT_LOG
         kill $SERVER_PID
         wait $SERVER_PID
+        set -e
```

Comment:
question: why do we need to move `set -e`? If it is before kill/wait it can help us detect non-graceful exits.

Reply:
Yes but at the same time on local runs it will exit before the server is properly killed. I am also a little hesitant on this. Since the tests are orthogonal to each other, I though it might make more sense to run everything and see which tests are failing instead of only failing on the first one

Reply:
As long as `set -e` is after `python ...` it would run all the tests. We need to either check for the return code of `wait` or move `set -e` before the `kill & wait` calls.

Reply:
@jbkyang-nvi friendly ping on this one.